kayenta-core/src/main/java/com/netflix/kayenta/canary/CanaryConfig.java

```diff
@@ -39,6 +39,8 @@
 
   @NotNull @Getter @Setter private String name;
 
+  @NotNull @Getter @Setter private String id;
```

Comment:
Had to add `id` to canary config, and it will be populated anytime the config is persisted.

It still does not _need_ to be provided on a POST or a PUT, so no changes are required to any dependent services.